app/controllers/api/public/base_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Api::Public::BaseController < ActionController::API
+  before_action :log_full_request, if: -> { Rails.application.secrets[:log_api_requests] }
+
+  private
+
+  def log_full_request
+    http_envs = {}.tap do |envs|
+      request.headers.each do |key, value|
+        envs[key] = value if key == key.upcase
+      end
+    end
+    Rails.logger.info("Headers: #{http_envs}")
```

Comment:
Would you delete the print statements? It's not necessary and ends up jamming paper trail